nomad/structs/acl.go

```diff
@@ -978,6 +978,24 @@ func (a *ACLAuthMethod) Validate(minTTL, maxTTL time.Duration) error {
 	return mErr.ErrorOrNil()
 }
 
+// Sanitize returns a copy of the ACLAuthMethod with any secrets redacted
+func (a *ACLAuthMethod) Sanitize() *ACLAuthMethod {
+	if a == nil {
+		return nil
+	}
+	// copy to ensure we do not mutate a pointer pulled directly out of state.
+	clean := a.Copy()
+	if clean.Config == nil {
+		return clean
+	}
```

Comment:
Can we skip the copy if there's no config?

```suggestion
	if a == nil || a.Config == nil {
		return a
	}

	clean := a.Copy()
```